include/boost/geometry/algorithms/detail/envelope/linestring.hpp

```diff
@@ -108,7 +116,9 @@ struct envelope_linear_on_spheroid
             Box segment_mbr;
             dispatch::envelope
                 <
-                    typename std::iterator_traits<iterator_type>::value_type
+                    typename std::iterator_traits<iterator_type>::value_type,
+                    0,
+                    DimensionCount
```

Comment:
I think I see your point here. It might be the case that `DimensionCount` should be replaced by `2` here.
I will double-check it.